src/backend/distributed/shared_library_init.c

```diff
@@ -733,6 +739,16 @@ RegisterCitusConfigVariables(void)
 		0,
 		NULL, NULL, NULL);
 
+	DefineCustomEnumVariable(
+		"citus.multi_shard_modify_executor",
```

Comment:
I think executor might cause confusion with router vs. real-time vs. task-tracker. Maybe `multi_shard_modify_mode`, or something that's more in line with postgres GUCs, e.g.: `enable_parallel_modification` (make it a boolean that defaults to true).